src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1144,25 +1144,24 @@ object Desugar {
           }
         }
 
-        case CompositeKind.Array(in.ArrayT(_, typ, addressability)) =>
+        case CompositeKind.Array(in.ArrayT(len, typ, addressability)) =>
           Violation.violation(addressability == Addressability.literal, "Literals have to be exclusive")
-          val indices = info.keyElementIndices(lit.elems)
-          val elems = lit.elems.zip(indices).map(e => (e._2, e._1.exp)).sortBy(_._1).map(_._2)
-          for { elemsD <- sequence(elems.map(e => compositeValD(ctx)(e, typ))) }
-            yield in.ArrayLit(typ, elemsD)(src)
+          for {
+            elemsD <- sequence(lit.elems.map(e => compositeValD(ctx)(e.exp, typ)))
+          } yield {
+            val mappedElemsD = info.keyElementIndices(lit.elems).zip(elemsD).toMap
+            val supplementedElemsD = Range.BigInt(0, len, 1).map(i => mappedElemsD.getOrElse(i, in.DfltVal(typ)(Source.Parser.Internal))).toVector
+            in.ArrayLit(typ, supplementedElemsD)(src)
+          }
```

Comment:
Why are you not using keyedElementsD here?

Reply:
Because array literals always have a fixed length, whereas the slice/sequence literals should be derived from the keyed elements.

For example, `[4]int { 2:12 }` is equivalent to `[4]int { 0, 0, 12, 0 }`, whereas `seq[int] { 2:12 }` is equivalent to `seq[int] { 0, 0, 12 }` (and the same for slices).

Reply:
One downside of this fix, is that for example `seq[int] { 100000:12 }` would generate a huge `Seq(...)` literal on the level of Viper...

Would you say that this is problematic? I don't expect such situations to some up very often.

Reply:
I do think that this is problematic. It would be nicer to handle this at the level of the encoding.

Reply:
However, you can also let it be for now. We could also discuss this in the gobra meeting.